File: src/main/java/net/felsstudio/fels/lib/Value.java
package main.java.net.felsstudio.fels.lib;

/**
 *
 * @author aNNiMON
 */
public interface Value extends Comparable<Value> {
    
    Object raw();
    
    int asInt();
    
    double asNumber();
    
    String asString();
    
    int type();
}

File: src/main/java/net/felsstudio/fels/lib/Value.kt
package main.java.net.felsstudio.fels.lib;

/**
 *
 * @author felek
 */
public interface Value extends Comparable<Value> {
    
    Object raw();
    
    int asInt();
    
    double asNumber();
    
    String asString();
    
    int type();
}
